source/slang/slang-serialize-types.h

```diff
@@ -0,0 +1,232 @@
+// slang-serialize-types.h
+#ifndef SLANG_SERIALIZE_TYPES_H
+#define SLANG_SERIALIZE_TYPES_H
+
+#include "../core/slang-riff.h"
+#include "../core/slang-string-slice-pool.h"
+#include "../core/slang-array-view.h"
+
+#include "slang-name.h"
+#include "slang-source-loc.h"
+
+namespace Slang {
+
+// Options for IR/AST/Debug serialization
+
+struct SerialOptionFlag
+{
+    typedef uint32_t Type;
+    enum Enum : Type
+    {
+        RawSourceLocation   = 0x01,     ///< If set will store directly SourceLoc - only useful if current source locs will be identical when read in (typically this is *NOT* the case)
+        DebugInfo           = 0x02,     ///< If set will output debug information, that can be reconstructed when read after being stored.
```

Comment:
I wonder if `DebugInfo` should more specifically be `SourceLocationInfo`, because that is really the main debugging-related information we emit today.

I guess the question is: if we added support for other debugging-related information, would we want it to all be controlled by the same flag, or would we want different flags for different kinds of debug information?